python/tank/platform/engine.py

```diff
@@ -1275,14 +1265,41 @@ def _do_invoke(self):
                         """
                         self._res = self._fn()
 
-                # Make sure that the invoker exists in the main thread:
-                invoker = Invoker()
-                if QtGui.QApplication.instance():
-                    invoker.moveToThread(QtGui.QApplication.instance().thread())
-                return invoker
+                class AsyncInvoker(QtCore.QObject):
```

Comment:
I am not a big fan of the class-inside-method approach. For KISS reasons, I think it should be avoided unless the class is very small. My personal opinion here is that this `AsyncInvoker` is in the gray area of what is acceptable - I would suggest moving this class out, place it below `Engine()` and call it `AsyncInvoker` - but having just written that, I realise it may not be possible due to the local QT imports - is that the reason the class is in-line? In that case, maybe we could add a small comment to highlight that.